File: html/old/backup/sacor2_2007/digitsums.cpp
// SACO 2nd Round 2007
// DigitSum solution by Bruce Merry
#include <iostream>

using namespace std;

int main()
{
    int n;
    cout << "Enter a positive integer: " << flush;
    cin >> n;
    cout << "Digit sum: ";
    if (n != 0 && n % 9 == 0) cout << "9\n";
    else cout << n % 9 << "\n";
    return 0;
}

File: html/old/backup/sacor2_2007/digitsums.py
#!/usr/bin/env python
# SACO 2nd Round 2007
# Solution by Marco Gallotta

N = raw_input("Enter a number: ")
while len(N) > 1:
    new_N = 0
    for digit in N:
        new_N += int(digit)
    N = str(new_N)
print "Digit sum:", N
